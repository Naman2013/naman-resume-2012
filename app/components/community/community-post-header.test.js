import React from 'react';
import renderer from 'react-test-renderer';
import jest from 'jest';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import CommunityPostHeader from './community-post-header';

it('it should render a header', () => {
  const testData = {
    errorOccurred: false,
    objectIconURL: '',
    showCreateNewPostButton: false,
    subtitleText: '',
    titleText: '',
  };
  const tree = renderer.create(
    <CommunityPostHeader
      {...testData}
    />,
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('it should render a defined header', () => {
  const testData = {
    errorOccurred: false,
    objectIconURL: '',
    showCreateNewPostButton: false,
    subtitleText: '',
    titleText: '',
  };
  const component = shallow(<CommunityPostHeader {...testData} />);
  expect(component).toBeDefined();
});

it('it should render a title and subtitle if one is passed', () => {
  const testData = {
    titleText: 'title',
    subtitleText: 'subtitle',
    objectIconURL: '',
    showCreateNewPostButton: false,
    errorOccurred: false,
  };
  const component = mount(<CommunityPostHeader {...testData} />);
  expect(component.find('.title-text').first().text()).toContain(testData.titleText);
  expect(component.find('.subtitle').first().text()).toContain(testData.subtitleText);
});
