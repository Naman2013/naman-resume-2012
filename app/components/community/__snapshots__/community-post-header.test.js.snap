// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render a header 1`] = `
<div
  className="communityPostHeader"
  data-jsx={190455770}
>
  <div
    className="left-container"
    data-jsx={190455770}
  >
    <div
      className="title"
      data-jsx={190455770}
    >
      <h1
        className="title-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
        data-jsx={190455770}
      />
    </div>
    <div
      className="subtitle"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      data-jsx={190455770}
    />
  </div>
  <div
    className="additional"
    data-jsx={190455770}
  >
    <div
      className="button-nav"
      data-jsx={190455770}
    >
      <span
        className="action-button"
        data-jsx={190455770}
      >
        <a
          className="button btn-primary"
          onClick={[Function]}
          style={Object {}}
        >
          Schedule Telescope
        </a>
      </span>
    </div>
  </div>
  <style
    jsx={true}
  >
    
      .communityPostHeader {
        font-family: brandon-grotesque, sans-serif;
        background: url(https://vega.slooh.com/assets/images/graphics/milkeyway_header.jpg) no-repeat;
        background-size: cover;
        min-height: 145px;
        display: inline-flex;
        flex-wrap: wrap;
        align-items: center;
        padding: 50px;
        justify-content: space-between;
        width: 100%;
      }

      .left-container {
        display: flex;
        flex-direction: column;
      }

      .title {
        display: inline-flex;
        align-items: center;
      }

      .subtitle {
        color: #fff;
        text-transform: none;
        font-weight: normal;
        margin-top: 5px;
      }

      .button-nav {
        text-align: left;
      }

      .title-text {
        margin: 0;
        color: #fff;
        text-transform: uppercase;
        font-weight: 600;
        font-size: 36px;
        padding-left: 15px;
      }

      svg {
        margin-right: 10px;
      }

      .sponsored {
        text-align: right;
        margin-right: 30px;
        color: $turqoise;
      }

      .action-button {
        margin-right: 30px;
      }

      @media(max-width:375px) {
        .title-text {
          font-size:28px
        }
      }
    
  </style>
</div>
`;
